jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/SignedMessage.java

```diff
@@ -0,0 +1,22 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe;
+
+public record SignedMessage(String message, String signature)
```

Comment:
Do we really need this class? Just to call a static method?

Reply:
This class is public. They might need to extend the authenticator to parse different message formats, because the spec doesn't specify how the message should be sent (right now being done as `application/x-www-form-urlencoded`). So I have a protected method which returns this `SignedMessage` class so they can customize it.

Reply:
But records cannot be extended???

Could this at least just be a protected inner class?

Reply:
There is no reason for this to be extended. There is reason to extend the authenticator to change how to the message/signature are parsed from the request.

I can move it to an inner class of EthereumAuthenticator though.

Reply:
Done.